code/clue.js

```diff
@@ -1,59 +1,350 @@
+    const roomList = weaponRooms[mystery.weapon.id];
+    mystery.room = shuffle(roomList, rooms);
+    document.getElementById("roomName").innerHTML = mystery.room;
+  } else {
+    mystery.room = randomSelector(rooms);
+    document.getElementById("roomName").innerHTML = mystery.room;
+  }
+};
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
+
+  document.getElementById("killerCard").style.background = mystery.killer.color;
+
+  document.getElementById("killerName").innerHTML =
+    mystery.killer.firstName + " " + mystery.killer.lastName;
+
+  document.getElementById("killerOccupation").innerHTML =
+    mystery.killer.occupation;
+
   document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+    "killerImage"
+  ).innerHTML = `<img src="${mystery.killer.image}">`;
+};
 
 // CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+  document.getElementById(
+    "weaponName"
+  ).innerHTML = `${mystery.weapon.name} which weight is ${mystery.weapon.weight}.`;
+};
+
+// LOADER
+
+document.getElementById("loaderItem").style.display = "none";
+
+// Invoke function
+
+document.getElementById("killerCard").onclick = () => {
+  loader();
+  pickKiller();
+};
+
+document.getElementById("weaponCard").onclick = () => {
+  loader();
+  pickWeapon();
+};
+document.getElementById("roomCard").onclick = () => {
+  loader();
+  // pickRoom();
+  shuffleFavRoom();
+};
+
+document.getElementById("mysteryCard").onclick = shuffleFavouriteWeapon;
+
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+// 'The murder was committed by Jacob Green, in the living room with a rope. '
+
+const revalMystery = () => {
+  document.getElementById(
+    "reveal"
+  ).innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} with a ${mystery.weapon.name}. Favorite wapon is ${mystery.favWeapon.id}`;
+};
+
+document.getElementById("reveal").onclick = () => {
+  revalMystery();
+  document.getElementById("loaderItem").style.display = "none";
+};
+
+// LOADER
+
+function loader() {
+  document.getElementById("loaderItem").style.display = "block";
+}
+
+// Defaoult function
+const reset = () => {
```

Comment:
Right now, there are some issues with the loader: 
- when I click on Reset before clicking on "Reveal the crime" the loader stays which confuses me as a user (like, "what is loading right now??")
- when have revealed a crime, but not clicked Reset, and then click on a card deck again, the loader reappears and there is no way to make it disappear, except clicking all the four card decks and then "Reveal the crime".

Including one line in the reset function could solve the issue: 
`  document.getElementById("loaderItem").style.display = "none";`

Reply:
Good point

Reply:
I updated the loader to work according the documentations. Now it stays active for a second after some card is clicked